tests/jballerina-unit-test/src/test/resources/test-src/javainterop/ballerina_types_as_interop_types.bal

```diff
@@ -160,6 +161,20 @@ public function interopWithAnydataReturn() returns boolean {
     return true;
 }
 
+public function interopWithHandleOrErrorReturn() {
+    handle|error handleOrError = acceptIntReturnIntArrayThrowsCheckedException(1);
+    test:assertEquals(handleOrError is error, false);
+    test:assertEquals(handleOrError is handle, true);
+
+    handleOrError = acceptIntReturnIntArrayThrowsCheckedException(5);
+    test:assertEquals(handleOrError is error, true);
+    error err = <error> handleOrError;
+    var message = err.detail()["message"];
+    string messageString = message is error ? message.toString() : message.toString();
```

Comment:
The if and else parts of `is error` check seems to be the same. Do we need this?

Reply:
Yes, It is needed right. IIUC this is a part of Ballerina's behaviour of error checking. `err.detail()["message"]` is belongs to `ballerina/lang.value:1.0.0:Cloneable` and and actually we are expecting a string or error here (in our case it would be a string). We can't directly apply `.toString` to a `var` or `string|error` directly. That's why error check is needed. But instead of `string messageString = message is error ? message.toString() : message.toString();`, we can use `string|error messageString = <string|error> message;` also.